File: ChatDemo/database/databasefactory.cpp
#include "databasefactory.h"
#include "globaldef.h"
#include "logindata/logindata.h"
#include "logdata/logdata.h"

/************         构造函数         *************/
DataBaseFactory::DataBaseFactory()
{

}

/************         创建数据库         *************/
DataBase *DataBaseFactory::createDataBase(int type)
{
    DataBase *dataBase = NULL;

    switch (type)
    {
    case GLOBALDEF::LOGINDATA: dataBase = new LoginData(); break;
    case GLOBALDEF::LOGDATA: dataBase = new LogData(); break;
    }

    return dataBase;
}

File: Server/db/databasefactory.py
#****************************************************
# 作者：YYC
# 功能：数据库简单工厂模式
# 日期：2017-11-02
#****************************************************

from globaldef import DATABASETYPE
from db.logdata import LogData

class DataBaseFactory():

    def createDataBase(self, type):
        dataBase = None

        # 日志数据库
        if(DATABASETYPE.LOGDATA == type):
            dataBase =  LogData()

        return dataBase